### PR TITLE
feat(growth): move issue details events to route analytics tracking

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr consolidates 2 high volume issue detail events to use the route analytics tracking. this will help manage the amplitude quote because instead of creating a separate event for each, we will combine them and send one for the route, while still maintaining all the data from before. 

GRW-744

### CODE REVIEW

File: static/app/views/organizationGroupDetails/groupDetails.tsx [hunk 1]
```diff
@@ -172,10 +170,10 @@ class GroupDetails extends Component<Props, State> {
 
   trackView(project: Project) {
     const {group, event} = this.state;
-    const {organization, params, location} = this.props;
+    const {params, location} = this.props;
     const {alert_date, alert_rule_id, alert_type} = location.query;
-    trackAdvancedAnalyticsEvent('issue_details.viewed', {
-      organization,
+
+    this.props.setRouteAnalyticsParams({
```
[the remark this answers is not in the document]
Comment (roggenkemper) [reply static/app/views/organizationGroupDetails/groupDetails.tsx:176]: oh sorry, i might've misunderstood you yesterday. will add the issue_details.viewed name back!

[the remark this answers is not in the document]
Comment (scefali) [reply static/app/views/organizationGroupDetails/groupDetails.tsx:177]: @roggenkemper yea the point is we want to keep the same event name but only for the issue details viewed event and not the other one. My bad if I didn't explain this well.

File: static/app/components/events/eventCause.tsx [hunk 2]
```diff
@@ -30,26 +30,13 @@ function EventCause({group, eventId, project, commitRow: CommitRow}: Props) {
...
+  useRouteAnalyticsParams({
     fetching,
-    committers.length,
-    project.id,
-    group?.id,
-    group?.issueCategory,
-  ]);
+    count: committers.length,
+    project_id: parseInt(project.id as string, 10),
+    group_id: parseInt(group?.id || '', 10),
```
Comment (scefali) [inline static/app/components/events/eventCause.tsx:37]: @roggenkemper I don't think we even need a lot of these fields if the are already being set in static/app/views/organizationGroupDetails/groupDetails.tsx

File: static/app/components/events/eventCause.tsx [hunk 3]
```diff
@@ -30,26 +30,13 @@ function EventCause({group, eventId, project, commitRow: CommitRow}: Props) {
...
-  }, [
-    organization,
+  useRouteAnalyticsParams({
     fetching,
-    committers.length,
-    project.id,
-    group?.id,
-    group?.issueCategory,
-  ]);
+    count: committers.length,
```
Comment (scefali) [inline static/app/components/events/eventCause.tsx:35]: @roggenkemper we probably should say `suspect_commit_count`

**Review by scefali — APPROVED** [review]

**Review by scttcper — APPROVED** [review]

**Pull request merged**